src/status_im/contexts/wallet/events.cljs

```diff
@@ -545,3 +549,53 @@
  :wallet/process-watch-only-account-from-backup
  (fn [_ [{:keys [backedUpWatchOnlyAccount]}]]
    {:fx [[:dispatch [:wallet/process-account-from-signal backedUpWatchOnlyAccount]]]}))
+
+(defn reconcile-watch-only-accounts
+  [_ [watch-only-accounts]]
+  {:fx (map (fn [account]
+              [:dispatch [:wallet/process-account-from-signal account]])
+            watch-only-accounts)})
+
+(rf/reg-event-fx :wallet/reconcile-watch-only-accounts reconcile-watch-only-accounts)
+
+(defn reconcile-keypairs
+  [{:keys [db]} [keypairs]]
+  (let [existing-keypairs-by-id               (get-in db [:wallet :keypairs])
+        existing-accounts-by-address          (get-in db [:wallet :accounts])
+        {:keys [removed-keypair-ids
+                removed-account-addresses
+                updated-keypairs-by-id
+                updated-accounts-by-address]} (data-store/reconcile-keypairs keypairs)
+        updated-keypair-ids                   (set (keys updated-keypairs-by-id))
+        updated-account-addresses             (set (keys updated-accounts-by-address))
+        existing-account-addresses            (set (keys existing-accounts-by-address))
+        new-account-addresses                 (clojure.set/difference updated-account-addresses
+                                                                      existing-account-addresses)
+        old-account-addresses                 (->> (vals existing-accounts-by-address)
+                                                   (filter (fn [{:keys [address key-uid]}]
+                                                             (and (contains? updated-keypair-ids key-uid)
+                                                                  (not (contains?
+                                                                        updated-accounts-by-address
+                                                                        address)))))
+                                                   (map :address))]
+    (cond-> {:db (-> db
+                     (assoc-in [:wallet :keypairs]
+                               (-> (apply dissoc existing-keypairs-by-id removed-keypair-ids)
+                                   (merge updated-keypairs-by-id)))
+                     (assoc-in [:wallet :accounts]
+                               (merge-with merge
+                                           (apply dissoc
+                                                  existing-accounts-by-address
+                                                  (into removed-account-addresses
+                                                        old-account-addresses))
+                                           updated-accounts-by-address)))}
```

Comment:
@smohamedjavid I've updated the account merging code here, let me know if you think this is okay 🙏